x-pack/plugins/fleet/dev_docs/developing_kibana_and_fleet_server.md

```diff
@@ -0,0 +1,278 @@
+        name: Fleet Server
+        id: fleet_server
+        inputs:
+          - type: fleet-server
+            keep_enabled: true
+            vars:
+              - name: host
+                value: 0.0.0.0
+                frozen: true
+              - name: port
+                value: 8220
+                frozen: true
+```
+
+1. Start up Kibana by running the following in another terminal session:
+
+```bash
+$ yarn start --ssl
+```
+
+1. Navigate to https://localhost:5601/app/fleet and click "Add Fleet Server"
+2. Select your localhost Fleet Server host and generate a service token
+3. Copy the service token somewhere convenient - you'll need it to run Fleet Server below
+
+## Start up Fleet Server
+
+These instructions assume you're currently in your local `fleet-server` checkout directory.
+
+1. Create a `fleet-server.dev.yml` file if one doesn't exist. This file is git ignored, so we can make our configuration changes directly instead of having to use environment variables or accidentally tracking changes to `fleet-server.yml`.
+
+```bash
+$ cp fleet-server.yml fleet-server.dev.yml
+```
+
+1. Update your `fleet-server.dev.yml` to look as follows
+
+```yaml
+# This config is intended to be used with a stand-alone fleet-server instance for development.
+output:
+  elasticsearch:
+    hosts: '<https://localhost:9200>'
+    # Copy the service token from the Fleet onboarding UI in Kibana
+    service_token: 'your_service_token'
+    # Fingerprint of the ca.crt certificate in Kibana's dev utils package
+    ssl.ca_trusted_fingerprint: 'f71f73085975fd977339a1909ebfe2df40db255e0d5bb56fc37246bf383ffc84'
+
+fleet:
+  agent:
+    id: '${FLEET_SERVER_AGENT_ID:dev-fleet-server}'
+
+inputs:
+  - type: fleet-server
+    policy.id: '${FLEET_SERVER_POLICY_ID:fleet-server-policy}'
+    server:
+      ssl:
+        enabled: true
+        certificate: ../kibana/packages/kbn-dev-utils/certs/fleet_server.crt
+        key: ../kibana/packages/kbn-dev-utils/certs/fleet_server.key
+        key_passphrase: ../kibana/packages/kbn-dev-utils/certs/fleet_server.key
+
+logging:
+  to_stderr: true # Force the logging output to stderr
+  pretty: true
+  level: '${LOG_LEVEL:DEBUG}'
+
+# Enables the stats endpoint under <http://localhost:5601> by default.
+# Additional stats can be found under <http://127.0.0.1:5066/stats> and <http://127.0.0.1:5066/state>
+http.enabled: true
+#http.host: <http://127.0.0.1>
+#http.port: 5601
+```
+
+```bash
+# Create standalone dev build
+$ DEV=true SNAPSHOT=true make release-darwin/amd64
+
+# Run dev build, provide your fingerprint and service token from before
+$ ELASTICSEARCH_CA_TRUSTED_FINGERPRINT="f71f73085975fd977339a1909ebfe2df40db255e0d5bb56fc37246bf383ffc84" ELASTICSEARCH_SERVICE_TOKEN="AAEAAWVsYXN0aWMvZmxlZXQtc2VydmVyL3Rva2VuLTE3MDQ0Njk3NTc4Mjk6dDhWaVl5Qy1SQ0taTjRRYzdibUx4UQ" FLEET_SERVER_POLICY_ID="fleet-server-policy" ./build/binaries/fleet-server-8.13.0-SNAPSHOT-darwin-x86_64/fleet-server -c fleet-server.yml
```

Comment:
nit: I usually just run `./build/binaries/fleet-server-8.13.0-SNAPSHOT-darwin-x86_64/fleet-server -c fleet-server.yml` and set the service token and other settings in the config file

Reply:
Good catch - I meant to update this since we put all that effort into writing config to `fleet-server.dev.yml` 😅. Updated!